### PR TITLE
Different versoning

### PR DESCRIPTION
**Pull request by shresthagrawal**
Solved issue #12 
I am still writing the tests.

### CODE REVIEW

File: release_bot/releasebot.py [hunk 1]
```diff
@@ -81,9 +81,24 @@ def find_open_release_issues(self):
             else:
                 for edge in reversed(edges):
                     cursor = edge['cursor']
-                    match = re.match(r'(.+) release', edge['node']['title'].lower())
+                    last_version = Version(self.github.latest_release())
```
Comment (jpopelka) [inline release_bot/releasebot.py]: `last_version` -> `latest_version`

('last' means there will be no more releases)

File: release_bot/releasebot.py [hunk 2]
```diff
@@ -81,9 +81,24 @@ def find_open_release_issues(self):
...
+                    if re_match:
+                        match = True
+                        version = re_match[1].strip()
+                    elif edge['node']['title'].lower().strip() == "new major release":
+                        match = True
+                        version = str(last_version.next_major())
+                    elif edge['node']['title'].lower().strip() == "new minor release":
+                        match = True
+                        version = str(last_version.next_minor())
+                    elif edge['node']['title'].lower().strip() == "new patch release":
```
Comment (jpopelka) [inline release_bot/releasebot.py]: Can we store `edge['node']['title'].lower().strip()` into a variable so we don't repeat it several times?

File: release_bot/releasebot.py [hunk 3]
```diff
@@ -124,12 +139,28 @@ def find_newest_release_pull_request(self):
...
+                    version = re_match[1].strip()
+                elif edge['node']['title'].lower().strip() == "new major release":
+                    match = True
+                    version = str(last_version.next_major())
+                elif edge['node']['title'].lower().strip() == "new minor release":
+                    match = True
+                    version = str(last_version.next_minor())
+                elif edge['node']['title'].lower().strip() == "new patch release":
+                    match = True
+                    version = str(last_version.next_patch())
```
Comment (jpopelka) [inline release_bot/releasebot.py]: The 2 blocks of code you're adding are the same, maybe they can be extracted into a method (e.g. process_version_from_title() - but name it as you like)

File: release_bot/releasebot.py [hunk 4]
```diff
@@ -124,12 +139,28 @@ def find_newest_release_pull_request(self):
 
             for edge in reversed(edges):
                 cursor = edge['cursor']
-                match = re.match(r'(.+) release', edge['node']['title'].lower())
-                if match and validate(match[1]):
+                last_version = Version(self.github.latest_release())
+                version = ''
+                match = False
+                re_match = re.match(r'(.+) release', edge['node']['title'].lower())
```
Comment (TomasTomecek) [inline release_bot/releasebot.py]: I don't we actually need this code here since the release PRs will not have these titles

Comment (jpopelka) [reply release_bot/releasebot.py]: how come?

Comment (TomasTomecek) [reply release_bot/releasebot.py]: The release pull requests won't be titled with "new major release", they will be titled with "0.4.2 release", still.

Comment (jpopelka) [reply release_bot/releasebot.py]: Ok, I thought you're talking just about  the `re.match` line.

**Comment by TomasTomecek** [comment]
Code LGTM, well done! We should document the feature now, please update README.

File: README.md [hunk 5]
```diff
@@ -5,7 +5,13 @@ Automate releases on Github and PyPi.
 ## Description
 
 This is a bot that helps maintainers deliver their software to users. It is meant to watch github repositories for
-release pull requests. The PR must be named in this format `0.1.0 release`. No other format is supported yet.
+release pull requests. The PR must be named in one of the following formats:
+* `0.1.0 release`
```
Comment (TomasTomecek) [inline README.md]:
```suggestion
* `0.1.0 release` if you want to create the "0.1.0" upstream release
```

File: README.md [hunk 6]
```diff
@@ -5,7 +5,13 @@ Automate releases on Github and PyPi.
...
+* `new major release` e.g. Changes Version 1.2.3 to 2.0.0
```
Comment (TomasTomecek) [inline README.md]:
```suggestion
* `new major release`, release-bot would then initiate a release from e.g. "1.2.3" to "2.0.0"
```

File: README.md [hunk 7]
```diff
@@ -5,7 +5,13 @@ Automate releases on Github and PyPi.
...
+* `new minor release` e.g. Changes Version 1.2.3 to 1.3.3
```
Comment (TomasTomecek) [inline README.md]:
```suggestion
* `new minor release` e.g. "1.2.3" to "1.3.0"
```

File: README.md [hunk 8]
```diff
@@ -5,7 +5,13 @@ Automate releases on Github and PyPi.
...
+* `new patch release` e.g. Changes Version 1.2.3 to 1.2.4
```
Comment (TomasTomecek) [inline README.md]:
```suggestion
* `new patch release` e.g. "1.2.3" to "1.2.4"
```

File: README.md [hunk 9]
```diff
@@ -5,7 +5,13 @@ Automate releases on Github and PyPi.
...
+
+No other format is supported yet.
```
Comment (TomasTomecek) [inline README.md]:
```suggestion
Release-bot now works with [SemVer](https://semver.org/) only.
```

**Comment by shresthagrawal** [comment]
@jpopelka Can you please help me with `rebase -i` command. I already tried a few times. I am unable to push anything if I use this command and everytime I had to forcepush (which I think so is not good). I tried to squash the unwanted commits but it doesn't removes/replaces those commits rather it adds more commits on the top of the Head. I also tried to fixup but It lead to something even more weird, my log started showing that I have contributed to 9day old commits already done by someone else. I somehow had a backup of the original git before rebase and I forcepushed it currently.

**Pull request closed without merging**